src/components/PaginationRow/PaginationRow.js

```diff
@@ -0,0 +1,199 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import { MenuItem, DropdownButton } from '../../index';
+
+const defaultMessages = {
+  firstPage: 'First Page',
+  previousPage: 'Previous Page',
+  nextPage: 'Next Page',
+  lastPage: 'Last Page',
+  perPage: 'per page',
+  of: 'of'
+};
+
+class PaginationRow extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.perPage = Number(props.perPage);
+    this.totalCount = Number(props.totalCount);
+    this.currentPage = Number(props.currentPage);
+
+    this.state = {
+      pageChangeValue: Number(props.currentPage)
+    };
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if (this.props.currentPage !== nextProps.currentPage) {
+      this.setState({ pageChangeValue: Number(nextProps.currentPage) });
+    }
+  }
+
+  msg(key) {
+    return this.props.messages[key] || defaultMessages[key];
+  }
+
+  totalPages() {
+    return Math.ceil(this.props.totalCount / this.perPage);
+  }
+
+  setPageRelative(diff) {
+    this.setPage(Number(this.props.currentPage) + diff);
+  }
+
+  setPage(page) {
+    if (page !== '') {
+      this.props.onPageSet(Number(page));
+    } else {
+      console.error("Page can't be blank");
+    }
+  }
+
+  handlePageChange(e) {
+    this.setState({ pageChangeValue: e.target.value });
+  }
+
+  handleFormSubmit(e) {
+    this.setPage(this.state.pageChangeValue);
+    e.preventDefault();
+  }
+
+  renderPerPageDropdown() {
+    const { perPageOptions, onPerPageSet } = this.props;
+
+    return (
+      <DropdownButton
+        dropup
+        bsStyle="default"
+        title={this.perPage}
+        id="per-page"
+      >
+        {perPageOptions.map(opt => {
+          return (
+            <MenuItem
+              active={opt === this.perPage}
+              onSelect={() => onPerPageSet(opt)}
+              key={opt}
+            >
+              {opt}
+            </MenuItem>
+          );
+        })}
+      </DropdownButton>
+    );
+  }
+
+  render() {
+    const perPageDropdown = this.renderPerPageDropdown();
+
+    const displayedRangeStart = (this.currentPage - 1) * this.perPage + 1;
+    const displayedRangeEnd = Math.min(
+      displayedRangeStart + this.perPage - 1,
+      this.totalCount
+    );
+    const displayedRange = `${displayedRangeStart}-${displayedRangeEnd}`;
+
+    const backButtonsClass = this.currentPage === 1 ? 'disabled' : '';
+    const nextButtonsClass =
+      this.currentPage * this.perPage >= this.totalCount ? 'disabled' : '';
+
+    const totalPages = this.totalPages();
+
+    return (
+      <div>
+        <form
+          className="content-view-pf-pagination clearfix"
```

Comment:
To make this usable with a table view, we need a mechanism to add `.table-view-pf-pagination` here.

Reply:
I added a `className` prop to `PaginationRow`